### DISCUSSION TITLE
blink1 listener client on Windows unable to connect to TA

### DISCUSSION THREAD

**Discussion by mcarlssen** [josephdadams/TallyArbiter#525] [General]
Apologies in advance if this is user error, but I'm only passingly familiar with python and not rock-solid on how TA works under the hood.

I'm setting up latest version of TA on a Win10 machine, which will host the TA server _and_ the listener service. I'm using OBS, latest version. I picked up a blink1 mk3 this week since that seemed the easiest way to keep the machine count down for such a simple configuration.

- Installed latest version of python, as well as manually installing the blink1, sio, and - surprisingly - zeroconf libraries.
- Blink1 is installed, working, and responding to Blink1Control2 app commands. 
- TA 2.0.16 is installed, OBS websocket plugin is installed, and the OBS source/device configurations are set and working properly. 

Downloaded latest `blink1-listener.py` script, ran it in elevated cmd as `> python3 blink1-listener.py --debug`. Based on what I see in the code, this should default to `localhost:4455`, which is correct in this instance (manually specifying either or both of the IP or port does not produce a change). Args print as expected. Then, nothing else happens.

Looking further in the code, it appears it's looping indefinitely around line 341 - apparently not having satisfied a condition in `TallyArbiterServerListener` or `ServiceBrowser`? I'm not sure which. No additional prompts or errors print, and of course TA does not see any new listeners.

I'm a bit stuck as I'm not sure where I strayed or what additional configuration the listener may be expecting from TA or other.

**Comment by mcarlssen** [comment]
I confirmed that the listener is indeed holding for conditions in the `True` loop, by adding an extraneous `print` following `sleep`, and observed the repeated print output after relaunching the listener.

[edit] `config.ini` shows the following parameters:
```
[DEFAULT]
deviceid = 
host = localhost
port = 4455
usemdns = True
clientuuid = 62030c23-c621-46fe-80d2-b46036f9bb2e
```

**Comment by josephdadams** [comment]
You’re using the blink listener client that only works with v3 of TA. If you want to run v2 of TA then you need this one: https://github.com/josephdadams/TallyArbiter-Blink1Listener

**Reply by mcarlssen** [reply]
Ah, of course. I did come across that file at one point during setup, but didn't connect the dots because I'm dense (and the archived repo threw me off a bit). Thanks for the assistance. Works an absolute treat.
